relay/blockchain/currency_network_proxy.py

```diff
@@ -158,7 +158,9 @@ def get_network_events(self, event_name: str, user_address: str=None, from_block
                 gevent.spawn(self.get_events, event_name, filter1, from_block),
                 gevent.spawn(self.get_events, event_name, filter2, from_block)
             ]
-            gevent.joinall(events, timeout=2)
+            gevent.joinall(events, timeout=10)
+            if events is None:
+                abort(504, 'Timeout fetching events')
             result = list(itertools.chain.from_iterable([event.value for event in events]))
```

Comment:
It is more likely that here the event.value is None if the greenlet didnt finish yet. This code is hard to understand because event and events are not events but a list of greenlets. Please rename to jobs or similar.